chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -393,4 +383,42 @@ private void switchToNodeBased() {
     }
   }
 
+
+  static {
+    factories.put(ImplementationEnum.NODE, new ImplementationFactory() {
```

Comment:
Allright, so we _do_ have the factory pattern here? Awesome! :tada: 

So we could actually allow plugins to register their own octree factories… but the enum prevents us from selecting them. Maybe use a string instead of the enum? :thinking:

Reply:
I've done that, using a string to select the factory to use and new factories can be registered using the `addImplementationFactory` method of `Octree`